backend/scripts/version.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+"""Create version information JSON file
+
+This file is created at build time and copied into the backend container so
+that we can include information that's not available at runtime.
+"""
+from datetime import datetime, timezone
+import json
+from pathlib import Path
+import sys
+import subprocess
+from typing import Optional
+
+
+def do(cmd: list[str]) -> list[str]:
+    result = subprocess.run(cmd, text=True, capture_output=True)
+    if result.returncode != 0:
+        raise Exception(f"Command {' '.join(cmd)!r} failed: {result.stderr}")
+    return [line.strip() for line in result.stdout.split("\n") if line.strip()]
+
+
+def getone(cmd: list[str], if_none: Optional[str] = None) -> str:
+    res = do(cmd)
+    if if_none is not None and len(res) == 0:
+        return if_none
+    if len(res) != 1:
+        raise Exception(f"Command {' '.join(cmd)!r} returned {len(res)} lines: {res}")
+    return res[0]
+
+
+def main():
+    top = Path(getone(["git", "rev-parse", "--show-toplevel"]))
+    backend = top / "backend"
```

Comment:
I thought `/` was not a supported operator for strings in python. What am I missing here?

Reply:
`top` isn't a string, it's a `Path`, and `Path` uses the `/` operator (which conveniently looks like a UNIX directory separator) to construct a new `Path` from the old one with a string tail element; `Path(".") / "foo" / "bar.py"` works out to `./foo/bar.py`.